pkg/reconciler/cmd/start.go

```diff
@@ -182,18 +181,7 @@ func Start(namespaced bool, managedNamespace string) {
 	// Watch Vertices with Generation changes (excluding scaling up/down)
 	if err := pipelineController.Watch(source.Kind(mgr.GetCache(), &dfv1.Vertex{}),
 		handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), &dfv1.Pipeline{}, handler.OnlyControllerOwner()),
-		predicate.And(
-			predicate.GenerationChangedPredicate{},
-			predicate.Funcs{
-				UpdateFunc: func(e event.UpdateEvent) bool {
-					if e.ObjectOld == nil || e.ObjectNew == nil {
-						return true
-					}
-					old, _ := e.ObjectOld.(*dfv1.Vertex)
-					new, _ := e.ObjectNew.(*dfv1.Vertex)
-					return !reflect.DeepEqual(new.Spec.WithOutReplicas(), old.Spec.WithOutReplicas())
-				}},
-		)); err != nil {
```

Comment:
Have to watch for replica update event as well for calculating the status of linked pod